client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -66,12 +66,50 @@
      */
     private final int maxConnectionsPerSegmentStore;
 
+    /**
+     * An internal property that determines whether TLS enabled is derived from Controller URI. It cannot be set
+     * directly by the caller. It is interpreted as:
+     *
+     * false - if {@link #enableTlsToController} and/or {@link #enableTlsToSegmentStore} is/are set, and if either
+     *         of them is set to false.
+     * true - if neither {@link #enableTlsToController} or {@link #enableTlsToSegmentStore} are set, or if both are
+     *        set to true.
+     */
+    private final boolean deriveTlsEnabledFromControllerURI;
+
+    /**
+     * An optional property representing whether to enable TLS for client's communication with the Controller.
+     *
+     * If this property and {@link #enableTlsToSegmentStore} are not set, and the scheme used in {@link #controllerURI}
+     * is {@code tls} or {@code pravegas}, TLS is automatically enabled for both client-to-Controller and
+     * client-to-Segment Store communications.
+     */
+    private final boolean enableTlsToController;
+
+    /**
+     * An optional property representing whether to enable TLS for client's communication with the Controller.
+     *
+     * If this property and {@link #enableTlsToController} are not set, and the scheme used in {@link #controllerURI}
+     * is {@code tls} or {@code pravegas}, TLS is automatically enabled for both client-to-Controller and
+     * client-to-Segment Store communications.
+     */
+    private final boolean enableTlsToSegmentStore;
```

Comment:
similar question as above.

Reply:
Sure, I have made the change in the latest commit. I am using Getter(AccessLevel.NONE).